x-pack/plugins/transform/server/routes/api/transforms.ts

```diff
@@ -238,36 +276,135 @@ const getTransforms = async (options: { transformId?: string }, callAsCurrentUse
   return await callAsCurrentUser('transform.getTransforms', options);
 };
 
+async function getIndexPatternId(
+  indexName: string,
+  savedObjectsClient: SavedObjectsClientContract
+) {
+  const response = await savedObjectsClient.find<IIndexPattern>({
+    type: 'index-pattern',
+    perPage: 10,
+    search: `"${indexName}"`,
+    searchFields: ['title'],
+    fields: ['title'],
+  });
+  const ip = response.saved_objects.find((obj) => obj.attributes.title === indexName);
+  return ip?.id;
+}
+
+async function deleteDestIndexPatternById(
+  indexPatternId: string,
+  savedObjectsClient: SavedObjectsClientContract
+) {
+  return await savedObjectsClient.delete('index-pattern', indexPatternId);
+}
+
 async function deleteTransforms(
   transformsInfo: TransformEndpointRequest[],
-  callAsCurrentUser: CallCluster
+  deleteDestIndex: boolean | undefined,
+  deleteDestIndexPattern: boolean | undefined,
+  ctx: RequestHandlerContext,
+  license: RouteDependencies['license'],
+  response: KibanaResponseFactory
 ) {
-  const results: TransformEndpointResult = {};
+  const tempResults: TransformEndpointResult = {};
+  const results: Record<string, DeleteTransformStatus> = {};
 
   for (const transformInfo of transformsInfo) {
+    let destinationIndex: string | undefined;
+    const transformDeleted: ResultData = { success: false };
+    const destIndexDeleted: ResultData = { success: false };
+    const destIndexPatternDeleted: ResultData = {
+      success: false,
+    };
     const transformId = transformInfo.id;
     try {
       if (transformInfo.state === TRANSFORM_STATE.FAILED) {
         try {
-          await callAsCurrentUser('transform.stopTransform', {
+          await ctx.transform!.dataClient.callAsCurrentUser('transform.stopTransform', {
             transformId,
             force: true,
             waitForCompletion: true,
           } as StopOptions);
         } catch (e) {
           if (isRequestTimeout(e)) {
             return fillResultsWithTimeouts({
-              results,
+              results: tempResults,
               id: transformId,
               items: transformsInfo,
               action: TRANSFORM_ACTIONS.DELETE,
             });
           }
         }
       }
+      // Grab destination index info to delete
+      try {
+        const transformConfigs = await getTransforms(
+          { transformId },
+          ctx.transform!.dataClient.callAsCurrentUser
+        );
+        const transformConfig = transformConfigs.transforms[0];
+        destinationIndex = Array.isArray(transformConfig.dest.index)
+          ? transformConfig.dest.index[0]
+          : transformConfig.dest.index;
+      } catch (getTransformConfigError) {
+        transformDeleted.error = wrapError(getTransformConfigError);
+        results[transformId] = {
+          transformDeleted,
+          destIndexDeleted,
+          destIndexPatternDeleted,
+          destinationIndex,
+        };
+        continue;
```

Comment:
do you need a `continue` statement?

Reply:
The continue is there to skip the rest of the work in the loop when we encounter an invalid transformId. Since we want to keep checking the rest of the transformIds in the array, I think the statement works well here.